subsys/net/ip/net_if.c

```diff
@@ -707,6 +696,17 @@ struct net_if_addr *net_if_ipv6_addr_add(struct net_if *iface,
 			net_sprint_ipv6_addr(addr),
 			net_addr_type2str(addr_type));
 
+		/* RFC 4862 5.4.2
+		 * "Before sending a Neighbor Solicitation, an interface
+		 * MUST join the all-nodes multicast address and the
+		 * solicited-node multicast address of the tentative address."
+		 */
+		/* The allnodes multicast group is only joined once as
```

Comment:
you can merge the 2 comments

Reply:
In principle yes, but those are two *different* comments, of different levels. To emphasize the difference, it's better to treat them as "annotations". First is normative reference annotation, applying to whole block of code (2 lines). 2nd is behavior-descriptive annotation, applying to just single statement. Merging two doesn't make much sense, and could lead to decreased clarity of plain confusion.

Reply:
true